### ISSUE TITLE
Images are not shown properly in the multimodal chatbot

### ISSUE THREAD

**Issue by hysts** [gradio-app/gradio#8963]
### Describe the bug

When passing `gr.MultimodalTextbox` to `gr.ChatInterface` so you can pass multiple images, images are only shown on the first `yield` of the chatbot and it's replaced with URLs afterwards.

(Also, `cache_examples=True` doesn't work. I opened  #8971 for this.)

### Have you searched existing issues?  🔎

- [X] I have searched and found no existing issues

### Reproduction

repro: https://huggingface.co/spaces/hysts-debug/multimodal-chat-image

```python
import time

import gradio as gr


def bot(message, history):
    out = []
    for word in "This is a test.".split():
        out.append(word)
        time.sleep(1)
        yield " ".join(out)


examples = [
    {
        "text": "hey",
        "files": ["cats.jpg", "dogs.jpg"],
    },
]

demo = gr.ChatInterface(
    fn=bot,
    multimodal=True,
    textbox=gr.MultimodalTextbox(
        file_types=["image"],
        file_count="multiple",
    ),
    examples=examples,
    cache_examples=False,
)
demo.queue().launch()
```

### Screenshot

https://github.com/user-attachments/assets/e83d4926-69f4-44aa-a536-09d74ad5bab2

### System Info

```shell
gradio==4.40.0
```

### Severity

Blocking usage of gradio

**Comment by Blaizzy** [comment]
Same here!

**Comment by pjsample** [comment]
I am having the same issue. Gradio==4.39

**Comment by hysts** [comment]
@freddyaboulton told me this issue would be fixed by #8847. I tested the wheel in the PR and it indeed fixes this issue.

**Comment by abidlabs** [comment]
Should be fixed by https://github.com/gradio-app/gradio/pull/8847, thanks @freddyaboulton!
